tailnet/conn.go

```diff
@@ -241,10 +263,18 @@ func NewConn(options *Options) (conn *Conn, err error) {
 	nodeUp.setAddresses(options.Addresses)
 	nodeUp.setBlockEndpoints(options.BlockEndpoints)
 	wireguardEngine.SetStatusCallback(nodeUp.setStatus)
-	wireguardEngine.SetNetInfoCallback(nodeUp.setNetInfo)
 	magicConn.SetDERPForcedWebsocketCallback(nodeUp.setDERPForcedWebsocket)
+	if telemetryStore != nil {
+		wireguardEngine.SetNetInfoCallback(func(ni *tailcfg.NetInfo) {
+			nodeUp.setNetInfo(ni)
+			telemetryStore.setNetInfo(ni)
+		})
+	} else {
+		wireguardEngine.SetNetInfoCallback(nodeUp.setNetInfo)
+	}
```

Comment:
How about this to avoid the double up:
```suggestion
	wireguardEngine.SetNetInfoCallback(func(ni *tailcfg.NetInfo) {
		nodeUp.setNetInfo(ni)
		if telemetryStore != nil {
			telemetryStore.setNetInfo(ni)
		}
	})
```

Reply:
It's not necessary to do the nil check each time the callback is run. Once set, `telemetryStore` can never be nil again.